lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2677,17 +2721,37 @@ CallInst *SPIRVToLLVM::transArbFloatInst(SPIRVInstruction *BI, BasicBlock *BB,
                    return ConstantInt::get(Int32Ty, Word);
                  });
 
-  FunctionType *FT = FunctionType::get(RetTy, ArgTys, false);
   std::string FuncName =
       SPIRVArbFloatIntelMap::rmap(OC) + getFuncAPIntSuffix(RetTy, ATy, BTy);
+  if (RetTy->getIntegerBitWidth() <= 64) {
+    FunctionType *FT = FunctionType::get(RetTy, ArgTys, false);
+    FunctionCallee FCallee = M->getOrInsertFunction(FuncName, FT);
+
+    auto *Func = cast<Function>(FCallee.getCallee());
+    Func->setCallingConv(CallingConv::SPIR_FUNC);
+    if (isFuncNoUnwind())
+      Func->addFnAttr(Attribute::NoUnwind);
+
+    return CallInst::Create(Func, Args, "", BB);
+  }
+
+  FunctionType *FT =
+      FunctionType::get(Type::getVoidTy(*Context), ArgTys, false);
   FunctionCallee FCallee = M->getOrInsertFunction(FuncName, FT);
 
   auto *Func = cast<Function>(FCallee.getCallee());
   Func->setCallingConv(CallingConv::SPIR_FUNC);
   if (isFuncNoUnwind())
     Func->addFnAttr(Attribute::NoUnwind);
 
```

Comment:
```suggestion

  if (RetTy->getIntegerBitWidth() <= 64)
    return CallInst::Create(Func, Args, "", BB);
```

Reply:
Thanks! Changed it in [5b525a8](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1254/commits/5b525a88aa6e1f1dd29eb71f2e8b493acc82837d)